internal/web_server/web/views/pages/dashboard_pages/edit_quiz.templ

```diff
@@ -253,23 +276,52 @@ script questionList() {
 		}
 }
 
-script changesSaved() {
-	document.body.addEventListener("htmx:afterSwap", (event) => {
-		// If the swap is successful, and it was not used to open the question modal
-		// Display a 'saved' message for 3 seconds
-		if (event.detail.xhr.status < 300 && event.target.id != 'question-modal' && event.target.id != 'alternatives-table') {
-			message = document.querySelector("#changes-saved-message")
+// Make the questions sortable, and save the new arrangement.
 
-			message.classList.remove("hidden")
-			message.classList.add("flex")
-			message.classList.add("animate-[fadeOpacity_3s_linear_forwards]")
+script sortableQuestions(quizId string) {
+	htmx.onLoad(function(content) {
+    Sortable.create(document.getElementById("question-list"), {
+			animation: 150,
+			handle: ".drag-handle",
+			onEnd: function(event) {
+				// Get all the question IDs and their arrangement in the list (first one is 1, second is 2, etc.)
+				// Send a POST request to the server with this mapping.
+				const questionList = document.getElementById("question-list");
+				const questions = questionList.children;
+				const questionIDs = Array.from(questions).map(question => question.getAttribute("data-question-id"));
+				questionIDs.shift();
 
-			setTimeout(
-				function() {
-					message.classList.add("hidden")
-					message.classList.remove("flex")
-					message.classList.remove("animate-[fadeOpacity_3s_linear_forwards]")
-				}, 3000)
-		}
-	})
+				const idMap = questionIDs.reduce((map, id, index) => {
+					map[index + 1] = id;
+					return map;
+				}, {});
+
+				fetch(`/api/v1/admin/quiz/rearrange-questions?quiz-id=${quizId}`, {
+					method: "POST",
+					headers: {
+						"Content-Type": "application/json",
+						"HX-Swap": "outerHTML",
+						"HX-Target": "#question-list",
+					},
+					body: JSON.stringify(idMap),
+				}).then((response) => {
+					let errorText = document.querySelectorAll(".error-question-list")[0];
+
+					if (response.status >= 400) {
```

Comment:
if there is an error, will the dragged item go back into its original place? or how does sortable work?
related to my other comment

Reply:
No, it does not.